yt/visualization/volume_rendering/scene.py

```diff
@@ -437,15 +443,49 @@ def save_annotated(
         """
         fname = self._setup_save(fname, render)
 
-        # which transfer function?
-        rs = self._get_render_sources()[0]
-        tf = rs.transfer_function
-        label = rs.data_source.ds._get_field_info(rs.field).get_label()
-
         ax = self._show_mpl(
             self._last_render.swapaxes(0, 1), sigma_clip=sigma_clip, dpi=dpi
         )
-        self._annotate(ax.axes, tf, rs, label=label, label_fmt=label_fmt)
+
+        # number of transfer functions?
+        num_trans_func = 0
+        for rs in self._get_render_sources():
+            try:
+                tf = rs.transfer_function
+                num_trans_func += 1
+            except AttributeError:
+                print("error")
+                pass
+
+        # which transfer function?
+        if num_trans_func == 1:
+            rs = self._get_render_sources()[0]
+            tf = rs.transfer_function
+            label = rs.data_source.ds._get_field_info(rs.field).get_label()
+            self._annotate(ax.axes, tf, rs, label=label, label_fmt=label_fmt)
+        else:
+            # set the origin and width and height of the colorbar region
+            if tf_rect is not None:
+                cbx0, cby0, cbw, cbh = tf_rect
+
+            if tf_rect is None:
+                cbx0 = 0.80
+                cby0 = 0.12
+                cbw = 0.12
+                cbh = 0.9
+
+            cbh_each = cbh / num_trans_func
+
+            for i, rs in enumerate(self._get_render_sources()):
+                ax = self._render_figure.add_axes(
+                    [cbx0, cby0 + i * cbh_each, 0.8 * cbw, 0.8 * cbh_each]
+                )
+                try:
+                    tf = rs.transfer_function
+                    label = rs.data_source.ds._get_field_info(rs.field).get_label()
+                    self._annotate_multi(ax, tf, rs, label=label, label_fmt=label_fmt)
+                except AttributeError:
+                    pass
```

Comment:
I suggest making the error catching narrower so we don't unintentionally catch anything else than we need
```suggestion
                try:
                    tf = rs.transfer_function
                except AttributeError:
                    pass
                else:
                    label = rs.data_source.ds._get_field_info(rs.field).get_label()
                    self._annotate_multi(ax, tf, rs, label=label, label_fmt=label_fmt)
```